### ISSUE TITLE
Static batching does not work when loading scenes additively

### ISSUE THREAD

**Issue by dexterdeluxe88** [playcanvas/engine#2455]
Forum Post: https://forum.playcanvas.com/t/static-batching-for-additive-scene-loading/15069
Sample Project: https://playcanvas.com/project/723629/settings

### Description

When additive loading a scene, the static batched models in this scene don't get loaded/rendered.

### Steps to Reproduce

1. Open sample project: https://playcanvas.com/project/723629/settings
2. Start from "Master" scene
3. Click "Load Van Scene"
4. Observe that just 1 van (with dynamic batching) gets loaded, but not the other van (with static batching enabled)

**Comment by leonidaspir** [comment]
Hi @dexterdeluxe88 ,

The pc.Batcher instance will fire automatically once when the pc.Application instance starts rendering.

Loading new scenes, which basically removes and adds entities in the hierarchy won't trigger that automatically. You can easily do this yourself, after the new scene hierarchy has been appended by running:

`
this.app.batcher.generate();
`
https://developer.playcanvas.com/en/api/pc.BatchManager.html#generate

**Comment by dexterdeluxe88** [comment]
@leonidaspir calling batcher.generate doesn't resolve the issue

**Comment by leonidaspir** [comment]
It seems the issue isn’t with batching, but with the **Static** model component flag. There is some optimization there that breaks batching. If you disable that flag batching works as expected:

![image](https://user-images.githubusercontent.com/29543178/94711691-bb4f5800-0348-11eb-92a5-3d97e6225d5b.png)

**Comment by mvaligursky** [comment]
yep, agreed with @leonidaspir - just what I wanted to write. The static flag is not needed for static batching I believe. I'll investigate in due time, but leave it off for now please.

**Comment by valentincognito** [comment]
I don't know if this is related or not but I noticed that batching and lightmapping doesn't work together at the moment.. I don't know if it's a normal behavior or not. I tried with static flag both enabled and disabled but it doesn't change anything.

If I just call:

``` javascript
this.app.batcher.generate();
```

I can see the batching working.

If I call batching then baking:
``` javascript
this.app.batcher.generate();
this.app.lightmapper.bake();
```

The light baking won't work

and finally if I call baking then batching:

``` javascript
this.app.lightmapper.bake();
this.app.batcher.generate();
```

Batching won't work. Seems like I cannot have the cake and eat it ^_^

Anyone ran into a similar problem ?

**Comment by mvaligursky** [comment]
@valentincognito - I don't have the answer to this, but it'd be worth investigating in the future. Could you please create a separate issue for it and link it to this one. If you have some simple test project to test on, that'd be fantastic as well.

**Comment by dexterdeluxe88** [comment]
> yep, agreed with @leonidaspir - just what I wanted to write. The static flag is not needed for static batching I believe. I'll investigate in due time, but leave it off for now please.

@mvaligursky do you have an estimate?

**Comment by mvaligursky** [comment]
> > yep, agreed with @leonidaspir - just what I wanted to write. The static flag is not needed for static batching I believe. I'll investigate in due time, but leave it off for now please.
> 
> @mvaligursky do you have an estimate?

I don't at the moment - as there seems to be an easy workaround (don't mark the mesh static), this does not sound like urgent at the moment.
